olad/Strerror_r.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Strerror_r.cpp
+ * Definition of strerror_r that is XSI-compliant.
```

Comment:
I'm generally slightly wary about the future maintainability of this, but I guess a pile of ifdef's to switch behaviour based on the detected version might not be any better.

Where did this come from, is there a link for credit/future reference?

Reply:
Oops. Forgot the copyright. I'll add that.